### PR TITLE
Idea to make serialization/mapping generic

### PR DESCRIPTION
**Pull request by Suamere**
Make serialization/mapping generic and configurable with base types. Required for use with custom value types. Current and new Unit Tests pass. Start review with DatabaseManager for clarity.  New unit tests/objects explain the required use case.

### CODE REVIEW

**Comment by Suamere** [comment]
I don't know what those CI Build errors are.  Review the code (if you can?) and let me know if anything can be fixed from me.  I haven't done many GitHub Pull Requests before, so these errors are foreign to me.

**Comment by psibernetic** [comment]
Hey Suamere, Travis CI isn't operational at the moment so don't worry about it, just an effort to get mono compatible. 

Sorry, I hadn't seen this sooner, I expected github to notify me of a pull request.

I will review the changes now.

File: src/Susanoo.Core/Mapping/Properties/ResultMappingBase.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        /// </summary>
+        private static bool CanConvert(Type from, Type to)
+        {
+            if (from == to) return true;
+            if (from.Namespace.Equals("system", StringComparison.OrdinalIgnoreCase) && to.Namespace.Equals("system", StringComparison.OrdinalIgnoreCase)) return false;
+            Func<Expression, UnaryExpression> bodyFunction = body => Expression.Convert(body, to);
+            ParameterExpression inp = Expression.Parameter(from, "inp");
+            try
+            {
+                Expression.Lambda(bodyFunction(inp), inp).Compile();
```
Comment (psibernetic) [inline src/Susanoo.Core/Mapping/Properties/ResultMappingBase.cs]: This may need a caching system to off-load performance hit.

**Comment by psibernetic** [comment]
Overall I really like the direction you went with this, CastValue has been something I avoided since inception. I'm going to accept the changes, thanks for the work!

**Comment by Suamere** [comment]
My pleasure.  Granted, it's selfish because I want to use my custom Value Objects with it.  I may give other small tidbits if I can come up with legitimate use case alterations. Thanks for reviewing it.

**Comment by psibernetic** [comment]
Good stuff, always glad to see support for more use cases. may I inquire how you found Susanoo and what you are using it on?

**Pull request merged**